### PR TITLE
311-script

### PR DESCRIPTION
**Pull request by rdlester**
Condenses and extracts 311 import into a script.

Adds "chunk csv parsing" mode to hpd_csv2sql.

Refactor of hpd_csv2sql to split csv cleaning and sql import into separate helpers.

infer_datetime_format set on pandas.to_datetime, hopefully speeding things up!

### CODE REVIEW

File: JPF/utils.py [hunk 1]
```diff
@@ -53,10 +53,10 @@ def mkdir_p(my_path):
...
 def connect():
 
-    user = os.environ['MYSQL_USER']
-    host = os.environ['MYSQL_HOST']
-    pw = os.environ['MYSQL_PASSWORD']
-    db = os.environ['MYSQL_DATABASE']
+    user = 'root'  # os.environ['MYSQL_USER']
+    host = 'localhost'  # os.environ['MYSQL_HOST']
+    pw = ''  # os.environ['MYSQL_PASSWORD']
+    db = 'heatseek'  # os.environ['MYSQL_DATABASE']
```
Comment (chrishenry) [inline JPF/utils.py]: Were you not able to set environment variables?

File: JPF/utils.py [hunk 2]
```diff
@@ -83,10 +83,28 @@ def guess_sqlcol(dfparam, max_col_len):
...
+        pickle_file - filename of pickle to save/load. should be a format in a unique folder if csv_chunk_size is specified.
+        db_action - "replace", "append"
+        truncate_columns - list of string-type columns to truncate
+        date_time_columns - list of date-type columns
+        sql_chunk_size - num of rows to send to sql in each operation
+        keep_cols - list of columns to keep in the database
+        max_col_len - length to truncate cells in truncate_columns to
+        date_format - expected format of dates in the table
+        csv_chunk_size - if defined, the number of rows to process from the CSV at a time.
+    """
```
Comment (chrishenry) [inline JPF/utils.py:107]: 👍

**Comment by chrishenry** [comment]
Nice, other than hardcoding your db config, this looks pretty awesome.

File: JPF/utils.py (hunk 1 above)
Comment (rdlester) [reply JPF/utils.py]: Yeah, I just haven't bothered with it yet, I do have this auto-env shell extension but didn't bother setting it up. Also this is a purely local instance so leaking it doesn't matter.

Removed.

**Pull request merged**